app/src/main/java/org/systers/mentorship/view/fragments/HomeFragment.kt

```diff
@@ -17,4 +35,47 @@ class HomeFragment : BaseFragment() {
     }
 
     override fun getLayoutResourceId(): Int = R.layout.fragment_home
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = DataBindingUtil.inflate(inflater, getLayoutResourceId(), container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        achievementsAdapter = AchievementsAdapter()
+        val linearLayoutManager = LinearLayoutManager(context)
+        val divider = DividerItemDecoration(context, linearLayoutManager.orientation)
+
+        rvAchievements.apply {
+            adapter = achievementsAdapter
+            layoutManager = linearLayoutManager
+            addItemDecoration(divider)
+        }
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        homeViewModel = ViewModelProviders.of(this).get(HomeViewModel::class.java)
+
+        with(homeViewModel) {
+            userStats.observe(viewLifecycleOwner, Observer { stats ->
+                binding.stats = stats
+                if (stats?.achievements?.isEmpty() != false) {
```

Comment:
This condition confuses me. `isEmpty()` function already returns a boolean, you can remove `!= false` inside if, and work with that logic.

Reply:
Initially I too did it the way you suggest, but Android Studio said that that way could cause a problem when the `stats` variable becomes null.

Let's look at it this way: if `stats` or `stats.achievements` is null, then the condition inside the if statement evaluates to null. An if statement requires a `Boolean`, but if we do things this way we would be passing it a `Boolean?`. To fix that, we can either add a non-null assertion `!!` operator to force the compiler to treat this expression as a non null Boolean, or we can use the elvis operator `?:` to supply an alternate value in case the expression evaluates to null.

Since using `!!` is unsafe, I used the elvis operator and my if statement looked this way:

`if (stats?.achievements?.isEmpty() ?: false)`

But when I do this, Android Studio suggests this:

`Equality check should be used instead of elvis for nullable boolean check`

And so we have an if statement that looks like this:

`if (stats?.achievements?.isEmpty() != false)`